designer/server/src/test/scala/pl/touk/nussknacker/ui/validation/UIProcessValidatorSpec.scala

```diff
@@ -1270,6 +1278,168 @@ class UIProcessValidatorSpec extends AnyFunSuite with Matchers with TableDrivenP
     result.warnings shouldBe ValidationWarnings.success
   }
 
+  test(
+    "validate Map service parameter based on additional config from provider - ValidationExpressionParameterValidator"
+  ) {
+    val process = createGraph(
+      List(
+        Source("inID", SourceRef(ProcessTestData.existingSourceFactory, List())),
+        Enricher(
+          "custom",
+          ServiceRef(
+            "mapParameterService",
+            List(
+              NodeParameter(
+                ParameterName("mapParam1"),
+                Expression.spel("""{'field': "valid", 'otherField': 'c'}""")
+              ),
+              NodeParameter(
+                ParameterName("mapParam2"),
+                Expression.spel("""{'field': "invalid", 'otherField': 'c'}""")
```

Comment:
Some cosmetic propositions:
1. You can use `.spel` string extension which is already imported
2. You can avoid tripple quotes by not using double quote ` " ` and use ` ' ` 
 
So here it could be  ` "{'field': 'invalid', 'otherField': 'c'}" `. Up to you. But please use consistent quotes - using ' ' in the same record in one field and " " in other can suggest that the ' ' is a char, but I think theres no difference. Please apply to other cases